source/code-snippets/usage-examples/updateOne.js

```diff
@@ -0,0 +1,39 @@
+// ignored first line
+const { MongoClient } = require("mongodb");
+
+// Replace the following with your MongoDB deployment's connection string.
+const uri =
+  "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
+
+const client = new MongoClient(uri, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+});
+
+async function run() {
+  try {
+    await client.connect();
+
+    const database = client.db("sample_mflix");
+    const collection = database.collection("movies");
+
+    // create a filter for a movie to update
+    const filter = { title: "Blacksmith Scene" };
+    const options = {};
+    options.upsert = true; // create a document if no documents match the filter
+    // create a document that sets the plot of the movie
+    const updateDoc = {
+      $set: {
+        plot:
+          "Blacksmith Scene is a silent film directed by William K.L. Dickson",
+      },
+    };
+    const result = await collection.updateOne(filter, updateDoc, options);
```

Comment:
Instead of a try around the entire body of this function, can we just use .next() and .catch() to handle the promise/errors generated by updateDoc? I'll probably create a PR updating the other methods too if you can get it to work here -- it's much closer to best practice than callbacks.

We'll want to leave the .catch() in the last line to handle connection-level errors though.

Reply:
+1 @terakilobyte on his thoughts for this

Reply:
As a compromise, could we leave the ``try``/``finally`` in here but add a

```
.then (console.log())
.catch(error) 
```
We need the finally in there anyway, so this might let us keep both models.

Reply:
I disagree -- I feel that handling error messages differently for different levels of abstraction demonstrates explicit error handling to users. Specifically I think it's useful to separate connection level errors from logic/internal errors related to the ``updateOne`` call itself; in a small example like this it doesn't make a huge difference, but it does matter for operations that use multiple methods in sequence where the result of one method affects the next call and so on. I also think it would be useful to explicitly print useful fields of the result object instead of just printing the entire object.

That being said, all of my thoughts here are really subjective. So if y'all feel like it's fine as is I think that's acceptable too -- just trying to make this as useful as possible for end users :smiling_imp: